### ISSUE TITLE
data sources (APIs, bs4 and Playwright scraping)

### ISSUE THREAD

**Issue by bbelderbos** [felipe74590/atx-events#1]
Please list 3-5 data sources you want to crawl to get the events source data. Then we can look at the best libraries to use.

**Comment by bbelderbos** [comment]
After #2 and for the first scraping effort open me a PR here:

```
git checkout -b event-scraping
# do work
git add .
git commit -m "..."
git push origin event-scraping
# open PR and assign me as reviewer
```

**Comment by felipe74590** [comment]
Data sources:
EventBrite API https://www.eventbrite.com/platform/docs/introduction 
Do512 https://do512.com/ 
Hey Austin https://heyaustin.com/austin-events/ 
What When Where Austin

**Comment by bbelderbos** [comment]
cool, sounds great, nice mix of scraping and APIs (latter is always the desired way because of more predictable / standardized outputs)

**Comment by bbelderbos** [comment]
@felipe74590 now that the first scraper works nicely, which one are you tackling next? New PR right?

**Comment by felipe74590** [comment]
Yes, I started working on the new data source and looking at how it's html is set up. Will open the new PR later today for Hey Austin webscraper source. https://heyaustin.com/austin-events/

**Comment by felipe74590** [comment]
@bbelderbos Started working on the EventBrite API data source but I found two other sites I would like to try to web scrape https://www.austintexas.org/events/ trying `events_soup = hot_soup.find_all("div", class_="contents")` but it returns an empty array. I think this is one of the sites that would require selenium or other web scraping plugins. Could you confirm if possible, just want to make sure I'm not doing it wrong. the content Beautiful soup returns mentions lots of scrips and event the inspect element on the page looks more complex.

**Comment by felipe74590** [comment]
This is the 4th site I was trying to scrape https://austin.culturemap.com/events/ but the following would return empty events_soup = hot_soup.find_all("div", class_="event-post")

**Comment by felipe74590** [comment]
After exploring Event Brite API, it looks like the features I need are deprecated, so I started to explore the Meta Graph APi https://developers.facebook.com/docs/graph-api/reference/event/

**Comment by bbelderbos** [comment]
> This is the 4th site I was trying to scrape https://austin.culturemap.com/events/ but the following would return empty events_soup = hot_soup.find_all("div", class_="event-post")

want me to check both? I can, busy coding today on platform v2, but will do before we meet, how is the fb api going?

**Comment by bbelderbos** [comment]
Nice find PredictHQ, chatgpt suggested that to me as well: 

<img width="795" alt="image" src="https://github.com/user-attachments/assets/d1a0a77a-d4b1-46b9-89f8-48499e206cd6">

**Comment by bbelderbos** [comment]
view-source:https://www.austintexas.org/events/

has rss feed - nice! https://www.austintexas.org/event/rss/

```
import feedparser

feed = feedparser.parse('https://www.austintexas.org/event/rss/')
for entry in feed.entries:
    print(entry.title)
    print(entry.link)
    print(entry.description)
    print('\n')
```

-> this works

**Comment by bbelderbos** [comment]
This one doesn't https://austin.culturemap.com/feeds/events.rss

But I do see events in the html / view source of https://austin.culturemap.com/events/?tags=20240920 -> e.g. "Scream Hollow Wicked Halloween Park" ?

**Comment by bbelderbos** [comment]
But yeah then soup is empty for that one:

```
import requests
import bs4

headers = {
    'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.3'
}
resp = requests.get('https://austin.culturemap.com/events/', headers=headers)
soup = bs4.BeautifulSoup(resp.text, 'html.parser')
events = soup.find_all('div', class_='event-post')  # []
```

**Comment by bbelderbos** [comment]
This works for me:

```
uv add playwright
uv run playwright install
```

Starter code:

```
from playwright.sync_api import sync_playwright

with sync_playwright() as p:
    with p.chromium.launch(headless=True) as browser:  # change to False if you want to see it in action
        page = browser.new_page()
        page.goto('https://austin.culturemap.com/events/?tags=20240920')
        page.wait_for_selector('.event-post')  # Wait for the specific selector to be present
        for event in page.query_selector_all("div.event-post"):
            print(event.inner_text())
```
